Mathlib/Topology/UniformSpace/Equicontinuity.lean

```diff
@@ -462,68 +825,168 @@ protected theorem EquicontinuousAt.closure {A : Set (X → α)} {x₀ : X} (hA :
   EquicontinuousAt.closure' (u := id) hA continuous_id
 #align equicontinuous_at.closure EquicontinuousAt.closure
 
-/-- If `𝓕 : ι → X → α` tends to `f : X → α` *pointwise* along some nontrivial filter, and if the
-family `𝓕` is equicontinuous at some `x₀ : X`, then the limit is continuous at `x₀`. -/
-theorem Filter.Tendsto.continuousAt_of_equicontinuousAt {l : Filter ι} [l.NeBot] {F : ι → X → α}
-    {f : X → α} {x₀ : X} (h₁ : Tendsto F l (𝓝 f)) (h₂ : EquicontinuousAt F x₀) :
-    ContinuousAt f x₀ :=
-  (equicontinuousAt_iff_range.mp h₂).closure.continuousAt
-    ⟨f, mem_closure_of_tendsto h₁ <| eventually_of_forall mem_range_self⟩
-#align filter.tendsto.continuous_at_of_equicontinuous_at Filter.Tendsto.continuousAt_of_equicontinuousAt
-
-/-- A version of `Equicontinuous.closure` applicable to subsets of types which embed continuously
-into `X → α` with the product topology. It turns out we don't need any other condition on the
-embedding than continuity, but in practice this will mostly be applied to `DFunLike` types where
-the coercion is injective. -/
+/-- If a set of functions is equicontinuous at some `x₀` within a set `S`, its closure for the
+product topology is also equicontinuous at `x₀` within `S`. This would also be true for the coarser
+topology of pointwise convergence on `S ∪ {x₀}`, see `EquicontinuousWithinAt.closure'`. -/
+protected theorem EquicontinuousWithinAt.closure {A : Set (X → α)} {S : Set X} {x₀ : X}
```

Comment:
Shouldn't this one be in the `Set.EquicontinuousWithinAt` namespace, since you're applying it to sets of functions? If you agree, this means you don't need to put on the prime on the other version in the `EquicontinuousWithinAt` namespace.
Same thing below for all the other variations.

Reply:
I'm not a big fan of this, because then `EquicontinuousWithinAt.closure` changes meaning if you're in namespace `Set` (opening `Set` is okay though), and the difference between the two is hard to tell when used with dot notation. Also, the current spelling does work fine with dot notation because `Set.EquicontinuousWithinAt` is an `abbrev`.

Reply:
My problem is more in the other direction: if you have an assumption `H` that something (which is not a set of functions, but rather a family of functions) is equicontinuous and you try `H.closure`, then this will fails with a weird error message. With my proposition, `H.closure` should do the right thing in any case.

Anyway, I will follow whatever you think is best.

Reply:
I think I would keep the current meaning of `closure'` and `closure` (because `closure` is more common and easier to use), but I would replace `EquicontinuousWithinAt.closure` with `Set.EquicontinuousWithinAt.closure` to minimize the risk that someones calls `H.closure` when `H` is about a family of functions instead of a set of functions. Of course, it's not perfect (as you say, in the namespace `Set`, it doesn't protect you against the mistake), but it would still be an improvement over the current situation. You have convinced me that having both named `closure` but separated only by the namespace would be too prone to confusion.

Reply:
Sounds good to me, I'll do that.